pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1160,7 +1113,62 @@ public void addTable(@Nonnull TableConfig tableConfig)
         throw new InvalidTableConfigException("UnSupported table type: " + tableType);
     }
 
-    handleBrokerResource(tableNameWithType, brokersForTenant);
+    String brokerTenantName = TagNameUtils.getBrokerTagForTenant(tableConfig.getTenantConfig().getBroker());
+    handleBrokerResource(tableNameWithType, HelixHelper.getInstancesWithTag(_helixZkManager, brokerTenantName));
+  }
+
+  /**
+   * Validates the tenant config for the table
+   */
+  private void validateTableTenantConfig(TableConfig tableConfig, String tableNameWithType, TableType tableType) {
+    TenantConfig tenantConfig = tableConfig.getTenantConfig();
+    if (tenantConfig == null || tenantConfig.getBroker() == null || tenantConfig.getServer() == null) {
+      throw new PinotHelixResourceManager.InvalidTableConfigException(
```

Comment:
Why do we need to explicitly specify the class as `PinotHelixResourceManager.InvalidTableConfigException`? Is it ok to just say `InvalidTableConfigException` since the exception is just in Class `PinotHelixResourceManager `?

Reply:
yes it should be okay. I havent written this method, moved the code into a method. But sure, I can clean that up too